src/RegistryCoordinator.sol

```diff
@@ -145,11 +147,21 @@ contract RegistryCoordinator is EIP712, Initializable, IRegistryCoordinator, ISo
      * @notice Registers msg.sender as an operator for one or more quorums. If any quorum reaches its maximum
      * operator capacity, this method will fail.
      * @param quorumNumbers is an ordered byte array containing the quorum numbers being registered for
+     * @param pubkeyRegistrationParams contains the G1 & G2 public keys of the operator, and a signature proving their ownership
+     * @dev the `pubkeyRegistrationParams` input param is ignored if the caller has previously registered a public key
      */
     function registerOperator(
         bytes calldata quorumNumbers,
-        string calldata socket
+        string calldata socket,
+        IBLSApkRegistry.PubkeyRegistrationParams calldata pubkeyRegistrationParams
     ) external onlyWhenNotPaused(PAUSED_REGISTER_OPERATOR) {
+        /**
+         * IF the operator has never registered a pubkey before, THEN register their pubkey
+         * OTHERWISE, simply ignore the provided `pubkeyRegistrationParams`
+         */
+        if (blsApkRegistry.operatorToPubkeyHash(msg.sender) == 0) {
+            blsApkRegistry.registerBLSPublicKey(msg.sender, pubkeyRegistrationParams);
+        }
         bytes32 operatorId = blsApkRegistry.getOperatorId(msg.sender);
```

Comment:
Since `getOperatorId` returns `operatorToPubkeyHash`, I think this is cleaner like this:

```
bytes32 operatorId = blsApkRegistry.getOperatorId(msg.sender);
if (operatorId == 0) {
  operatorId = blsApkRegistry.registerBLSPublicKey(msg.sender, params);
}
```

Reply:
agreed, don't know why I thought this reverted if the operator wasn't registered or something. will make this change

Reply:
changed here. good callout https://github.com/Layr-Labs/eigenlayer-middleware/pull/100/commits/db000925f55ba3bca2db005008fb55feab68e60b